stdlib/buffer.mli

```diff
@@ -167,18 +167,16 @@ val add_subbytes : t -> bytes -> int -> int -> unit
 val add_substitute : t -> (string -> string) -> string -> unit
 (** [add_substitute b f s] appends the string pattern [s] at the end
    of buffer [b] with substitution.
-   The substitution process looks for variables into
-   the pattern and substitutes each variable name by its value, as
+   The substitution process looks for variable references in
+   the pattern and substitutes each variable reference with its value, as
    obtained by applying the mapping [f] to the variable name. Inside the
-   string pattern, a variable name immediately follows a non-escaped
-   [$] character and is one of the following:
+   string pattern, a variable reference is a non-escaped [$] immediately
+   followed by a variable name, which is one of the following:
```

Comment:
> The way the `$` character is escaped is inconsistent with escaping in other places (eg. OCaml strings, format strings) as it's impossible to escape the `\` character.

That's true, it's this way because I wanted to make minimal changes to the behavior of the function.

> It's impossible to have this function write `\$` without using a trick

That's in fact fixed by this PR. Look at the test file, the function maps:
`\$` to `$`
`\\$` to `\$`
`\\\$` to `\\$`
(and so on). It's a different quoting convention, but it **is** surjective, unlike the current version of `add_substitute`.

Reply:
Actually, what is impossible without tricks is a backslash followed by a variable reference. We could fix this by switching to the usual quoting convention, but I'm afraid this would entail a lot of breakage.